### PR TITLE
Compass: added COMPASS_OFFS_MAX parameter

### PR DESCRIPTION
**Pull request by tridge**
this is useful for aircraft that have hatch magnets. It avoids having to disable arming checks

### CODE REVIEW

**Comment by tridge** [comment]
this was flight tested on Sunday, no issues

File: libraries/AP_Compass/CompassCalibrator.h [hunk 1]
```diff
@@ -22,7 +22,7 @@ class CompassCalibrator {
 
     CompassCalibrator();
 
-    void start(bool retry=false, float delay=0.0f);
+    void start(bool retry=false, float delay=0.0f, uint16_t offset_max=600);
```
Comment (OXINARF) [inline libraries/AP_Compass/CompassCalibrator.h]: Why the default value? I think the parameter should always be used, so having a default doesn't look right to me.

**Comment by tridge** [comment]
thanks, fixed both issues and re-pushed over PR branch

**Comment by OXINARF** [comment]
Merged, thanks!

P.S.: Who knew I would be merging a PR from @tridge? :smile:

**Pull request merged**